### PR TITLE
[fix](alter inverted index) add or drop inverted index also need change table state to SCHEMA_CHANGE

### PR DESCRIPTION
**Pull request by Tanya-W**
# Proposed changes

Issue Number: close #xxx

## Problem summary

before this pr, add or drop inverted index not change table state, maybe multiple alter jobs executed at the same time, that may lead to some unexpected problems.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by Tanya-W** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java [hunk 1]
```diff
@@ -897,6 +903,10 @@ private void replayPendingJob(SchemaChangeJobV2 replayedJob) throws MetaNotFound
         olapTable.writeLock();
         try {
             addShadowIndexToCatalog(olapTable);
+            if (invertedIndexChange) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java:906]: All schema change job will set table's state to SCHEMA_CHANGE.
Why we have a `if` here?

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.04 seconds
 stream load tsv:          471 seconds loaded 74807831229 Bytes, about 151 MB/s
 stream load json:         37 seconds loaded 2358488459 Bytes, about 60 MB/s
 stream load orc:          75 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230306151631_clickbench_pr_109459.html

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java (hunk 1 above)
Comment (Tanya-W) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java:903]: othre shcema change job set table's state to SCHEMA_CHANGE in function `replayCreateJob`, invertedIndexChange job begin with WAITING_TXN, no PENDING state, so here need set table state to SCHEMA_CHANGE when replay job begin at WAITING_TXN

**Comment by Tanya-W** [comment]
run feut

**Comment by Tanya-W** [comment]
run p0

**Comment by Tanya-W** [comment]
run p0

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**